Sources/_StringProcessing/Compiler.swift

```diff
@@ -108,11 +108,86 @@ class Compiler {
       // We stick quoted content into read-only constant strings
       builder.buildMatchSequence(q.literal)
 
+    case .atom(let a) where a.assertionKind != nil:
+      try emitAssertion(a.assertionKind!)
+
     case .customCharacterClass, .atom:
       throw unsupported(node._dumpBase)
     }
   }
 
+  func emitAssertion(_ kind: AST.Atom.AssertionKind) throws {
+    // FIXME: Depends on API model we have... We may want to
+    // think through some of these with API interactions in mind
+    //
+    // This might break how we use `bounds` for both slicing
+    // and things like `firstIndex`, that is `firstIndex` may
+    // need to supply both a slice bounds and a per-search bounds.
+    switch kind {
```

Comment:
(please ignore until you're back from break)

@timvermeulen @natecook1000 . We have a bounds for getting around the slicing problem, and we have a bounds in which we search, but we might need to make those two separate concepts for anchors like `^`, which match at the beginning of the input. If we're always advancing our bounds as part of `firstIndex`, then `^` will _always_ match if it's key-ed off of that.